packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -114,3 +114,7 @@ function writeServerCookie (event: H3Event, name: string, value: any, opts: Cook
     // else ignore if cookie doesn't exist in browser and value is null/undefined
   }
 }
+
+function prepareStructuredClone (value: MaybeRef<any>): any {
+  return JSON.parse(JSON.stringify(toRaw(value)))
+}
```

Comment:
what about instead literally using the decode/encode option of the cookie?

Reply:
We could do that. Though, I think since this is specifically for making the payload encodable for structuredClone, it could introduce unexpected problems when a user supplies an encoder that isn't compatible. 

We'd also have to throw an error if the encode or decode functions are undefined, which is theoretically possible.

I'd probably not open up that error source. Though we could also make `prepareStructuredClone` user definable in the CookieOptions.

Reply:
I don't think this will open the error source as the encoder/decoder are used anyway to save/read the cookie. We should use the defaults if the user doesn't supply a custom encoder/decoder.

Reply:
Makes sense. Made that change, though I had to move the default encoder and decoder to separate functions as typescript could not infer `CookieDefaults.encode` and `CookieDefaults.decode` to be always defined.

Reply:
I think we could use `!` or make an assertion on L34 instead.

Reply:
pushed a refactor to use `satisfies` to avoid need for `!` entirely.